### PR TITLE
Update process-related docs

### PR DESCRIPTION
**Pull request by rshym**
- naming.md: add the rule to not move tags 
- releases.md: how to handle self-references in projects 
- README.md: add Xen-troops process section 
- Add description for the inter-repo dependencies

### CODE REVIEW

File: process/naming.md
Comment (lorc) [inline process/naming.md]: not exactly. Update may change public API, so major part of the tag should be increased. Jus say something like

"new release tag should be created according to release rules and SemVer requirements"

File: process/releases.md
Comment (lorc) [inline process/releases.md]: Well, this is a tricky one as there will be moment in time where we will have invalid moulin file because the commit is merged by tag still does not exist.  CI will fail for sure. Any unlucky user who might want to build the project at that exact moment will get an error... 

This is not an issue if we are using branches instead of tags. But we need tags for releases. I can't come with better solution right now, so probably we'll have to left this as is.

File: process/inter-repo.md
Comment (lorc) [inline process/inter-repo.md]: "This rule is especially important if you work in a team."

File: process/inter-repo.md
Comment (lorc) [inline process/inter-repo.md]: s/not intended/unintended

File: process/inter-repo.md
Comment (lorc) [inline process/inter-repo.md]: Maybe make it level 2 header?

File: process/inter-repo.md
Comment (lorc) [inline process/inter-repo.md]: Maybe make it level 2 header?

File: process/naming.md
Comment (rshym) [reply process/naming.md]: Fixed.

File: process/inter-repo.md
Comment (rshym) [reply process/inter-repo.md]: Fixed.

File: process/inter-repo.md
Comment (rshym) [reply process/inter-repo.md]: Fixed.

File: process/inter-repo.md
Comment (rshym) [reply process/inter-repo.md]: Done.

File: process/inter-repo.md
Comment (rshym) [reply process/inter-repo.md]: Done.

**Comment by rshym** [comment]
Comments are addressed. Ready for re-review.

**Review by lorc — APPROVED** [review]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

@klogg, may I ask you to take a look?

File: process/releases.md
Comment (oleksiimoisieiev) [inline process/releases.md]: I think version will have to include some product abbr. for example: we had zephyr release for rz_a2m with tag/branch say - rz_g2m-v0.1.0
And then on top of it we have released v0.1.0 but for rz_g3s so we will call it rz_g3s-v0.1.0 and have no problem with duplications

File: process/naming.md
Comment (oleksiimoisieiev) [inline process/naming.md]: that's why branch will save us from too much extra work

File: process/releases.md
Comment (rshym) [reply process/releases.md]: Reworked. Now we use release branches instead of non-existing tags.

File: process/releases.md
Comment (rshym) [reply process/releases.md]: The comment is addressed in the previous commit for the naming.md:
- described that we may need to use permanent platform-specific branches in some cases
- release tags are created with the according platform prefixes.

**Comment by rshym** [comment]
I have updated `naming.md` and `releases.md`.
Please re-review.

File: process/releases.md
Comment (lorc) [inline process/releases.md]: I am no sure I get this. Basically, you are saying that this YAML will track v2.3 branch with all its updates. How we can use this for release? It is not fixed.

Comment (rshym) [reply process/releases.md]: Yes. You are right. The problem is that we can't use the fixed commit or tag here, because we have to know the commit hash that doesn't exist before the merge,OR we have to use the tag that can be created only after the merge.

Comment (lorc) [reply process/releases.md]: I am wondering how repo-based and west.yaml-based projects tackle this issue...

Comment (rshym) [reply process/releases.md]: To be honest, we have one more way. To merge commit with everything fixed in recipes and in the yaml, except self-reference.  And, after that, we have to merge the commit with the self-reference pointing to the just merged commit. And to use that last commit as release one.
In this case, we will have properly working-compilable set of commits, but with one issue.
During the fetch of the sources, we will have some un-syncronization. Because we will use yaml from some release tag, and after that we will get `meta-xt-super-product` fetched to another commit that is `parent` to the release commit.
This will work, but will introduce some confusion.

Comment (rshym) [reply process/releases.md]: Maybe I am stuck in something, but the only way to do things properly - is to move yaml-configuration for the `meta-xt-super-product` to separate repo `xt-super-product` that will contain just one file - yaml-configuration. And there we will be able to create proper and clear releases.

Comment (rshym) [reply process/releases.md]: After your comment, I looked into https://github.com/xen-troops/zephyr-dom0-xt/blob/rpi5-v0.1.0/west.yml.
And I think I found the root cause and how to solve it. Our problem is already mentioned self-reference inside yaml.

Putting aside how to implement a proposed solution right now...

If we remove
```
    - type: git
      url: https://github.com/xen-troops/meta-xt-super-product.git
      rev: v2.3
```
from the `super-product.yaml`, we can merge and release properly.
During the release we just fix things, merge the commit, and put release tag on it.
When we need to build product in it's current or release state, we fetch HEAD or specific commit/tag and run moulin.
After the fetch, we have the product in the required state, and all its dependencies are described in yaml.

However, I do not know how to properly handle the situation with the fetched product that need to be placed into the expected hierarchy of build folders.

File: process/releases.md
Comment (klogg) [inline process/releases.md]: ... and require management review & approval.

**Comment by klogg** [comment]
For the textual part (except YAML, discussion of which eventually may introduce more changes that I will need to re-review), with one minor comment above: `Reviewed-by: Artem Mygaiev <artem_mygaiev@epam.com>`

File: process/releases.md
Comment (rshym) [reply process/releases.md]: Added.

File: process/releases.md
Comment (rshym) [reply process/releases.md]: I have removed part of the text with specific handling of the moulin-based projects, because there is no need for the special handling after the review and merge https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/143

**Comment by rshym** [comment]
Comments are addressed.
- " and require management review & approval." is added;
- special handling of the moulin-based projects is removed

After the latest comment, only the second commit was modified "releases.md: describe release branches".

Ready for the re-review.

**Comment by oleksiimoisieiev** [comment]
This approach looks good to me
`Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>`

**Comment by lorc** [comment]
s/naming.md: updates on tags and branched/naming.md: updates on tags and branche**s**

**Review by lorc — APPROVED** [review]
With that small comment about the commit header:

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
The misprint in the first commit message is fixed.
Reviewers are added.
Should be merged only after the https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/143

**Comment by klogg** [comment]
Please proofread and fix language. With that done
```
Reviewed-by: Artem Mygaiev <artem_mygaiev@epam.com>
```

**Pull request merged**
